### DISCUSSION TITLE
[Proposal] Changes to RenderEntityControls

### DISCUSSION THREAD

**Discussion by juriroemer** [Arkisto-Platform/describo-online#50] [General]
As I'm working on figuring out which of my changes for RDS I can bring upstream, I would like to propose a few tweaks to the layout of RenderEntityControls, I'm going to divide this into a few parts:

--- 
#### 1. if `isRootDataset`:

1. Remove (`v-if`) the "Load Root Dataset" button instead of disabling it
2. Swap the positions and types of the "Add Property" and "Save Crate as Template" buttons.
    - Reason: "Add Property" is the main action here and will be used way more often than "Save Crate as Template", therefore it should be on the right and of type `primary`, not the other way around.

---

#### 2. if `!isRootDataset`

1. Rename "Load Root Dataset" to something like "⬅ Back To Root"
    - Reason: I think the meaning of "Load Root Dataset" is quite hard to understand without a deeper knowledge of the software and JSON-LD
2. Swap Positions of "Delete Entity" and "Add Property" buttons, swap types of "Save Entity Template" and "Add Property" button
    - Reason: same as in # 2 above

---

#### 3. tweaks to `AddPropertyDialog` and `SaveCrateAsTemplate`

1. `AddPropertyDialog`: Move close button to the upper right instead of lower right
2. `SaveCrateAsTemplate`:
    - Add close button to upper right to stay consistent with `AddPropertyDialog`
    - Change `Save` button to type `primary` 

Example of how this could look, minus the save-as-primary (this is what I did for Sciebo RDS, upstream could of course keep the current color scheme with the blue backgrounds instead of a border):

`AddPropertyDialog`:
![image](https://user-images.githubusercontent.com/31971585/180007635-297ccc40-95cf-4a71-b2a6-1902f0ae6d81.png)

`SaveCrateAsTemplate`:
![image](https://user-images.githubusercontent.com/31971585/180007291-1a58ccca-36ad-4fba-9678-443c094ae462.png)

Disclaimer: the `Add Property` and `Save Crate as Template` buttons in the screenshots do not conform # 1 and # 2 of this Proposal.

**Comment by marcolarosa** [comment]
> #### 1. if `isRootDataset`:
> 
>     1. Remove (`v-if`) the "Load Root Dataset" button instead of disabling it

My admittedly limited understanding of usability is that buttons shouldn't really appear and disappear. Buttons should be disabled when the context changes and they are not required. Hence why it's written the way it is.

But I'm not an expert in this area by far so if you can provide some information supporting your idea i'd be happy to accept the change.

> 
>     2. Swap the positions and types of the "Add Property" and "Save Crate as Template" buttons.
>        
>        * Reason: "Add Property" is the main action here and will be used way more often than "Save Crate as Template", therefore it should be on the right and of type `primary`, not the other way around.

I'd be happy for add property to become a primary but the location on the left (rather than the right edge) is b/c it's the primary action so it follows the principle of left aligned menu bars. Also, as the window gets bigger, having it on the right means a lot of mouse moving to get to the right side of the screen to add property and then come back to operate in the dialog. My preference is that it stays where it is.

> 
> 
> #### 2. if `!isRootDataset`
> 
>     1. Rename "Load Root Dataset" to something like "⬅ Back To Root"
>        
>        * Reason: I think the meaning of "Load Root Dataset" is quite hard to understand without a deeper knowledge of the software and JSON-LD
> 

This is tough. You have a valid argument but root dataset means something quite important and arguably, users should know at least that when creating metadata. Maybe others have an opinion on this.

>     2. Swap Positions of "Delete Entity" and "Add Property" buttons, swap types of "Save Entity Template" and "Add Property" button
>        
>        * Reason: same as in # 2 above

Same reason as above to leave as is. I expect add property will be used often - hence keeping to the left and delete / save less so - hence keeping them far away on the right.
> 
> 
> #### 3. tweaks to `AddPropertyDialog` and `SaveCrateAsTemplate`
> 
>     1. `AddPropertyDialog`: Move close button to the upper right instead of lower right

Good idea - will definitely take it.
> 
>     2. `SaveCrateAsTemplate`:
>        
>        * Add close button to upper right to stay consistent with `AddPropertyDialog`
>        * Change `Save` button to type `primary`

I don't follow this. save crate as template is a primary button ...?
> 
> 
> Example of how this could look, minus the save-as-primary (this is what I did for Sciebo RDS, upstream could of course keep the current color scheme with the blue backgrounds instead of a border):
> 
> `AddPropertyDialog`: ![image](https://user-images.githubusercontent.com/31971585/180007635-297ccc40-95cf-4a71-b2a6-1902f0ae6d81.png)
> 
> `SaveCrateAsTemplate`: ![image](https://user-images.githubusercontent.com/31971585/180007291-1a58ccca-36ad-4fba-9678-443c094ae462.png)
> 
> Disclaimer: the `Add Property` and `Save Crate as Template` buttons in the screenshots do not conform # 1 and # 2 of this Proposal.
